eng/ci/public-build.yml

```diff
@@ -13,17 +13,13 @@ pr:
     include:
     - v4.x
     - release_4.0
-    - v3.x
-    - release_3.0
```

Comment:
Noticed the v3 branch names. Removed them as well. V3 has its own pipeline files in v3 branches.